docs/docs/forms.mdx

```diff
@@ -361,75 +361,151 @@ to fully customize the validation process.
 If none of the predefined [Slot Mappings](./forms.mdx#slot-mappings) fit your use
 case, you can use the
 [Custom Action](./actions.mdx#custom-actions) `validate_{form_name}` to write your own
-extraction code. Rasa Open Source will trigger this function when the form is run.
+extraction code. Rasa Open Source will trigger this action when the form is run.
+
+If you're using the Rasa SDK we recommend you to extend the provided
+`FormValidationAction`. When using the `FormValidationAction`, two things are needed to
+extract customs slots:
+
+1. Override `required_slots`. It should return a list of slot names which your form
+   requires to be filled.
+2. Define a method `extract_<slot_name>` for every custom slot which you want to
+   extract.
 
-Make sure your custom action returns `SlotSet` events for every extracted value.
 The following example shows the implementation of a custom slot mapping which sets
-a slot based on the length of the last user message.
+the slot `outdoor_seating` based on whether the keyword `outdoor` was present in the
+last user message.
 
 ```python
-from typing import Dict, Text, List
+from typing import Dict, Text, List, Optional, Any
 
 from rasa_sdk import Tracker
-from rasa_sdk.events import EventType
 from rasa_sdk.executor import CollectingDispatcher
-from rasa_sdk import Action
-from rasa_sdk.events import SlotSet
+from rasa_sdk.forms import FormValidationAction
 
 
-class ValidateRestaurantForm(Action):
+class ValidateRestaurantForm(FormValidationAction):
     def name(self) -> Text:
         return "validate_restaurant_form"
 
-    def run(
+    async def required_slots(
+        self,
+        dispatcher: "CollectingDispatcher",
+        tracker: "Tracker",
+        domain: "DomainDict",
+    ) -> Optional[List[Text]]:
+        return ["outdoor_seating"]
+
+    async def extract_outdoor_seating(
         self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
-    ) -> List[EventType]:
+    ) -> Dict[Text, Any]:
         text_of_last_user_message = tracker.latest_message.get("text")
+        sit_outside = "outdoor" in text_of_last_user_message
 
-        return [SlotSet("user_message_length", len(text_of_last_user_message))]
+        return {"outdoor_seating": sit_outside}
 ```
 
-### Requesting Extra Slots
+By default the `FormValidationAction`
+will automatically set the [`requested_slot`](forms.mdx#the-requested_slot-slot) to the
+first slot specified in `required_slots` which is not filled.
```

Comment:
I think the above comment is still relevant. Could we mention what's the default behaviour, i.e. when the user doesn't override `required_slots`?

Reply:
> that if the user doesn't return anything from required_slots (or doesn't override required_slots at all), then the form won't request any further slots.

If `required_slots` returns `[]` it will indeed deactivate the form.

>  Could we mention what's the default behaviour, i.e. when the user doesn't override required_slots?

You meant that Rasa Open Source will then request the next slot  based on the slot mappings defined in the domain?

Reply:
> My understanding was that if I just want to implement custom slot mapping, I implement the corresponding extract_{slot}, nothing more.

That **only** works if this slot was mapped in the domain!

> And if I just want to change the set of required slots, I override required_slots and possibly add extract_{slot} for slots that aren't listed for this form in my domain file.

Exactly.

Reply:
> If `required_slots` returns `[]` it will indeed deactivate the form.

Ah, that's fair now with the new way `required_slot` will be implemented in `FormValidationAction`.

> You meant that Rasa Open Source will then request the next slot based on the slot mappings defined in the domain?

Exactly.

> That only works if this slot was mapped in the domain!

Ah, right, sure. And we plan to throw a warning if the user writes a custom extractor for a slot that's already mapped in the domain, right? So, then the only recommended way to do custom mapping would be to **not** map that slot in the domain, write a custom extractor method, and add the slot to the list inside `required_slots`, right?

Reply:
> You meant that Rasa Open Source will then request the next slot based on the slot mappings defined in the domain?

> Exactly.

Will do 👍 

> And we plan to throw a warning if the user writes a custom extractor for a slot that's already mapped in the domain, right? 

Can do 👍 

> So, then the only recommended way to do custom mapping would be to not map that slot in the domain, write a custom extractor method, and add the slot to the list inside required_slots, right?

Sounds good 👍